### DISCUSSION TITLE
Socket session is not working on ubuntu server but local is working.

### DISCUSSION THREAD

**Discussion by 0xdanieltran** [socketio/socket.io#4989] [General] [answered]
export const configureSocketMiddleware = (io: IOServer) => {
  io.engine.use(sessionMiddleware);
  io.use((socket: any, next: any) => {
    (socket.request.session as any).reload((err: any) => {
      if (err) socket.disconnect();
      else {
        next();
      }
    });
  });
};

I can pass this middleware on localhost but once I move the node.js project to server, I couldn't use the session.

**Comment by darrachequesne** [comment, marked as answer]
Hi!

If the session is not found, then it might indicate that the cookie is not sent due to CORS. Is that your case?

Reference: https://socket.io/docs/v4/handling-cors/

**Reply by 0xdanieltran** [reply]
Thx for your replying.
Yeah. cookie is not sending vis socket from local (http) frontend to server (https).
Cookie is only sending on local both.

**Reply by 0xdanieltran** [reply]
Hi
I can't solve the problem still

**Reply by darrachequesne** [reply]
@aiden77mori that's weird! Could you please share your configuration? You can fork this basic project: https://github.com/socketio/socket.io-fiddle
